src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceFacadeImpl.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.cwrdapi.service.impl;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.CaseWorkerDomain;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.CaseWorkerProfile;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.ServiceRoleMapping;
+import uk.gov.hmcts.reform.cwrdapi.controllers.internal.impl.CaseWorkerInternalApiClientImpl;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkersProfileCreationRequest;
+import uk.gov.hmcts.reform.cwrdapi.service.CaseWorkerProfileConverter;
+import uk.gov.hmcts.reform.cwrdapi.service.CaseWorkerServiceFacade;
+import uk.gov.hmcts.reform.cwrdapi.service.ExcelAdaptorService;
+import uk.gov.hmcts.reform.cwrdapi.service.ExcelValidatorService;
+import uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants;
+
+import java.util.List;
+
+@Service
+@Slf4j
+public class CaseWorkerServiceFacadeImpl implements CaseWorkerServiceFacade {
+
+    @Autowired
+    ExcelValidatorService excelValidatorService;
+
+    @Autowired
+    ExcelAdaptorService excelAdaptorService;
+
+    @Autowired
+    CaseWorkerInternalApiClientImpl caseWorkerInternalClient;
+
+    @Autowired
+    ValidationService validationService;
+
+    @Autowired
+    CaseWorkerProfileConverter caseWorkerProfileConverter;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public ResponseEntity<Object> processFile(MultipartFile file) {
+        long time1 = System.currentTimeMillis();
+        Workbook workbook = excelValidatorService.validateExcelFile(file);
+        log.info("----Time taken to validate the given file "
```

Comment:
you can print file name in loggers of time taken

Reply:
sure